modules/core/src/testFixtures/java/org/apache/ignite/internal/testframework/IgniteTestUtils.java

```diff
@@ -322,6 +326,35 @@ public static Throwable assertThrowsWithCause(
         throw new AssertionError("Exception has not been thrown.");
     }
 
+    /**
+     * Checks the given {@link IgniteException} that it is the instance of the correct class, has the correct code and message.
+     * If the exception is incorrect, {@link AssertionError} is thrown and the incorrect exception is added to suppressed exceptions.
+     *
+     * @param expectedClass Expected class of the exception.
+     * @param expectedErrorCode Expected error code of the {@link IgniteException}.
+     * @param expectedMessageSubstring A substring that must be contained in the error message of the exception.
+     * @param exception Exception to check.
+     */
+    public static void assertExceptionIsExpected(
```

Comment:
I'd expected that it'll be an overloaded version of `assertThrowsWithCause` with code parameter in addition to cls and msg that will use extended `hasCause` with extra code param
```
    public static boolean hasCause(
            Throwable t,
            Class<?> cls,
            @Nullable Integer errorCode
            @Nullable String messageFragment
    ) {
```

Reply:
done

Reply:
It's not "with cause" because it checks the exact type of the exception, and it seems to be more natural because if we assume that the exception has some error code, it should be the exception of some desired type and we don't have to check it's cause.

Reply:
Ok, you are right.